src/main/java/com/google/cloud/solutions/datalineage/extractor/BigQueryTableCreator.java

```diff
@@ -99,7 +99,7 @@ public static BigQueryTableEntity fromLinkedResource(String linkedResource) {
 
   private static final String BQ_RESOURCE_FORMAT =
       String.format(
-          "^projects/(?<%s>%s)/datasets/(?<%s>%s)/tables/(?<%s>%s)$",
+          "^projects/(?<%s>%s)/datasets/(?<%s>%s)/tables/(?<%s>%s)(\\$[0-9]+)?$",
```

Comment:
to make it consistent and easy for use can you make this as following:

```java
private static final String PARTITION_ID_TAG  =  "partitionId";
private static final String PARTITION_PATTERN = "[0-9]+";
...
private static final String BQ_RESOURCE_FORMAT =
String.format(
          "^projects/(?<%s>%s)/datasets/(?<%s>%s)/tables/(?<%s>%s)(?<%s>\\$%s)?$",
          PROJECT_ID_TAG, PROJECT_PATTERN, DATASET_ID_TAG, DATASET_PATTERN, TABLE_ID_TAG,
          TABLE_PATTERN, PARTITION_ID_TAG, PARTITION_PATTERN);
```

Reply:
I initially had the partition constant extracted in this manner, but switched to the inline regex update because the constant wasn't extracted like the other patterns constants (the behavior is more like the ^projects/ matching component, for example). I don't feel strongly about this though and make this change. At least with the extracted constant it's clear that partitions are numeric, and I can see how this is easier to follow.